src/services/similarity.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @async
+ * @param {number} recipient_id - The ID of the recipient.
+ * @param {boolean} [cluster] - Specifies whether to cluster the results. Default value is false.
+ * @returns {Promise<Array>} A promise that resolves to an array of similar goals.
+ */
+// eslint-disable-next-line import/prefer-default-export
+export async function similarGoalsForRecipient(recipient_id, cluster) {
+  /**
+    * response without clustering looks like:
+    * result: [
+      *  {
+        *    goal1: {
+          *      id: 1, name: "Identify strategies", grantId: 1,
+          *    },
+        *    goal2: {
+          *      id: 2, name: "Identify strategies", grantId: 1,
+          *    },
+        *    similarity: 0.921823748234,
+        *  },
+      *  {
+        *    goal1: {
+          *      id: 1, name: "Identify strategies", grantId: 2,
+          *    },
+        *    goal2: {
+          *      id: 2, name: "Identify strategies", grantId: 2,
+          *    },
+        *    similarity: 0.921823748234,
+        *  },
+      * ]
+    */
+  try {
+    const { SIMILARITY_ENDPOINT } = process.env;
+    const response = await fetch(
```

Comment:
I am very out of the loop I guess, I had no idea fetch was in node core now

Reply:
yeah! I don't remember when it happened (16? 18?) but I'm happy it did